main/src/ca/uwaterloo/flix/util/Result.scala

```diff
@@ -123,6 +123,26 @@ object Result {
     Ok(res.toList)
   }
 
+  /**
+    * Applies f to each element in the list and flattens the results.
+    *
+    * Fails at the first error found, or returns the new list.
+    */
+  def flatTraverse[T, S, E](xs: Iterable[T])(f: T => Result[List[S], E]): Result[List[S], E] = {
```

Comment:
@mlutze @stephentetley Does this function have a more common name?

I don't think we have it in the Flix standard library. I guess its like traverse followed by map with flatten?